crates/iota-indexer/src/db.rs

```diff
@@ -311,11 +363,86 @@ pub mod setup_postgres {
             let store = PgIndexerStore::new(blocking_cp, indexer_metrics.clone());
             return Indexer::start_writer(&indexer_config, store, indexer_metrics).await;
         } else if indexer_config.rpc_server_worker {
+            check_db_migration_consistency(&mut conn)?;
             return Indexer::start_reader(&indexer_config, &registry, db_url.to_string()).await;
         } else if indexer_config.analytical_worker {
             let store = PgIndexerAnalyticalStore::new(blocking_cp);
             return Indexer::start_analytical_worker(store, indexer_metrics.clone()).await;
         }
         Ok(())
     }
+
+    #[cfg(feature = "pg_integration")]
+    #[cfg(test)]
+    mod tests {
+        use diesel::{
+            migration::{Migration, MigrationSource},
+            pg::Pg,
+        };
+        use diesel_migrations::MigrationHarness;
+        use secrecy::Secret;
+
+        use crate::{
+            db::setup_postgres::{self, MIGRATIONS},
+            test_utils::TestDatabase,
+        };
+
+        fn database_url(db_name: &str) -> Secret<String> {
+            secrecy::Secret::new(format!(
+                "postgres://postgres:postgrespw@localhost:5432/{db_name}"
+            ))
+        }
```

Comment:
This shares functionality with `common::get_indexer_db_url`, and is best to consolidate the two in `test_utils`.

Let's create a new issue to handle this.

Reply:
issue created: https://github.com/iotaledger/iota/issues/7224